repository/repository-server/src/main/java/org/eclipse/vorto/repository/core/IModelRepository.java

```diff
@@ -127,4 +127,32 @@
 	 * @return
 	 */
 	Optional<FileContent> getFileContent(ModelId modelId, String fileName);
-}
+	
+	/**
+	 * Attaches the given file to the model
+	 * 
+	 * @param modelid The modelId where to attach the file
+	 * @param fileName the filename
+	 * @param content the content of the file
+	 * @param userContext the user context
+	 * @return
+	 */
+	boolean attachFile(ModelId modelid, String fileName, byte[] content, IUserContext userContext);
```

Comment:
You can use FileContent class to pass to this function instead of filename and content individually. The other methods behave the same.